test/Microsoft.AspNetCore.Razor.Evolution.Test/IntegrationTests/RazorIRNodeWriter.cs

```diff
@@ -147,7 +147,10 @@ protected void WriteName(RazorIRNode node)
 
         protected void WriteLocation(RazorIRNode node)
         {
-            _writer.Write(node.SourceLocation.ToString());
+            if (node.DocumentLocation != null)
```

Comment:
MappingLocation isn't a struct currently. When working on the code gen we frequently check if `SourceLocation` is set on a particular `RazorIRNode` which makes it difficult to differentiate between `SourceLocation` 0,0,0 and one just not being set. We'd have to force all source locations to initially be set to -1,-1,-1 and then check against that repeatedly.

Long story short I feel that having a nullable document location works well here; especially given how much IRNodes are passed around.